WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,181 @@
+          </div>
+        <% end %>
+      </div>
+    <% else %>
+      <div class="row competition-list">
+        <div class="col-xs-12 col-md-12">
+          <div id="comps-map"></div>
+        </div>
+      </div>
+    <% end %>
+  <% end %>
 </div>
+
+<script>
+// See https://github.com/cubing/worldcubeassociation.org/issues/356
+function mapSize() {
+  var nav = parseInt($(".navbar").outerHeight(true));
+  var head = parseInt($("h2").outerHeight(true));
+  var form = parseInt($(".form-group").outerHeight(true));
+  var foot = parseInt($(".footer").outerHeight(true));
+
+  var height = parseInt($(window).innerHeight());
+
+  var map = height - nav - head - form - foot;
+
+  $("#comps-map").height( map );
+}
+
+function affixWidth() {
+  // From https://github.com/twbs/bootstrap/issues/6350#issuecomment-16069663
+  // also http://stackoverflow.com/a/20301984
+  // and http://stackoverflow.com/a/24908611
+  var affixBase = $(".container");
+  var width = affixBase.width() - parseInt(affixBase.css('padding-left')) - parseInt(affixBase.css('padding-right'));
+  $("#simple-nav").width(width);
+}
+
+// From http://stackoverflow.com/a/30966965
+$(".js-scroll-to").click(function(e) {
+  e.preventDefault();
+
+  // Only apply the movement if there is a div to go to.
+  if(!$(this).hasClass('disabled')) {
+    // Little hack to activate the affix thing and
+    // move correctly to the anchor.
+    $("#simple-nav").removeClass('affix-top').addClass('affix');
+
+    var destination = $(this).attr('href');
+
+    $('html, body').animate({
+      scrollTop: $(destination).offset().top - 40
+    }, 0);
+  }
+});
+
+$(function() {
+  <% if params[:display] == 'Map' %>
+    var competitions = <%= @competitions.map do |c|
+        {
+          id: c.id,
+          name: c.name,
+          latitude_degrees: c.latitude_degrees,
+          longitude_degrees: c.longitude_degrees,
+          cityName: c.cityName,
+          marker_date: c.start_date.to_formatted_s(:long),
+          is_over: c.is_over?,
+          url: competition_path(c)
+        }
+      end.to_json.html_safe %>;
+
+    $("#comps-map").competitionsMap(competitions);
+
+    mapSize();
+
+    $(window).resize(function() {
+      mapSize();
+    });
+  <% end %>
+
+  $(".disabled").each(function() {
+    $(this).removeAttr('href');
+  });
+
+  if (!isMobile()) {
+    // From http://www.sitepoint.com/understanding-bootstraps-affix-scrollspy-plugins/
+    $("#simple-nav").affix({
+      offset: {
+        top: $("#simple-nav").offset().top
+      }
+    });
+  }
+
+  if (isBigScreen()) {
+    $('.col-md-12').addClass('wide');
+  }
+
+  if (isMediumScreen()) {
+    $('.col-md-12').addClass('medium');
+  }
+
+  // Keep the navigation bar width (see above)
+  affixWidth();
+});
+
+// We check for small screens and remove the affix functionality
+$(window).resize(function() {
+  if (isBigScreen()) {
```

Comment:
I may be misunderstanding this stuff, but I'm pretty sure we can acheive what's going on here with pure css by using `@media` queries. Hit me up on gchat and lets talk about this.